diff --git a/drivers/power/mediatek/battery_common_fg_20.c b/drivers/power/mediatek/battery_common_fg_20.c
index 7082d4f..3df5857 100644
--- a/drivers/power/mediatek/battery_common_fg_20.c
+++ b/drivers/power/mediatek/battery_common_fg_20.c
@@ -127,6 +127,7 @@ bool tpd_usb_charging_flag = true;
 /* Battery Logging Entry */
 /* ////////////////////////////////////////////////////////////////////////////// */
 int Enable_BATDRV_LOG = BAT_LOG_CRTI;
+extern unsigned int g_fg_battery_id;	//add by jiahao for print battery id
 
 /* ///////////////////////////////////////////////////////////////////////////////////////// */
 /* // Smart Battery Structure */
@@ -2214,6 +2215,7 @@ void mt_battery_GetBatteryData(void)
 		BMT_status.charger_type, BMT_status.SOC, BMT_status.UI_SOC, BMT_status.UI_SOC2,
 		g_bcct_flag, get_usb_current_unlimited(), get_bat_charging_current_level() / 100,
 		BMT_status.IBattery / 10);
+	battery_log(BAT_LOG_CRTI,"Battery id = %d \n", g_fg_battery_id);	//add by jiahao for print battery id
 }
 
 
diff --git a/drivers/power/mediatek/battery_meter_fg_20.c b/drivers/power/mediatek/battery_meter_fg_20.c
index b25537a..b4e9564 100644
--- a/drivers/power/mediatek/battery_meter_fg_20.c
+++ b/drivers/power/mediatek/battery_meter_fg_20.c
@@ -507,6 +507,7 @@ kal_bool gFG_Is_offset_init = KAL_FALSE;
 
 #ifdef MTK_MULTI_BAT_PROFILE_SUPPORT
 unsigned int g_fg_battery_id = 0;
+EXPORT_SYMBOL(g_fg_battery_id);	//add by jiahao for print battery id
 
 #ifdef MTK_GET_BATTERY_ID_BY_AUXADC
 void fgauge_get_profile_id(void)
